examples/spline_example.cpp

```diff
@@ -0,0 +1,287 @@
+{
+  URCL_LOG_INFO("Starting joint-based trajectory forward");
+  g_my_driver->writeTrajectoryControlMessage(urcl::control::TrajectoryControlMessage::TRAJECTORY_START, p_p.size());
+
+  for (size_t i = 0; i < p_p.size() && p_p.size() == time.size() && p_p[i].size() == 6; i++)
+  {
+    // MoveJ
+    if (!use_spline_interpolation_)
+    {
+      g_my_driver->writeTrajectoryPoint(p_p[i], false, time[i]);
+    }
+    else  // Use spline interpolation
+    {
+      // QUINTIC
+      if (p_v.size() == time.size() && p_a.size() == time.size() && p_v[i].size() == 6 && p_a[i].size() == 6)
+      {
+        g_my_driver->writeTrajectorySplinePoint(p_p[i], p_v[i], p_a[i], time[i]);
+      }
+      // CUBIC
+      else if (p_v.size() == time.size() && p_v[i].size() == 6)
+      {
+        g_my_driver->writeTrajectorySplinePoint(p_p[i], p_v[i], time[i]);
+      }
+      else
+      {
+        g_my_driver->writeTrajectorySplinePoint(p_p[i], time[i]);
+      }
+    }
+  }
+  URCL_LOG_INFO("Finished Sending Trajectory");
+}
+
+// We need a callback function to register. See UrDriver's parameters for details.
+void handleRobotProgramState(bool program_running)
+{
+  // Print the text in green so we see it better
+  std::cout << "\033[1;32mProgram running: " << std::boolalpha << program_running << "\033[0m\n" << std::endl;
+}
+
+// Callback function for trajectory execution.
+bool g_trajectory_running(false);
+void handleTrajectoryState(control::TrajectoryResult state)
+{
+  // trajectory_state = state;
+  g_trajectory_running = false;
+  std::string report = "?";
+  switch (state)
+  {
+    case control::TrajectoryResult::TRAJECTORY_RESULT_SUCCESS:
+      report = "success";
+      break;
+    case control::TrajectoryResult::TRAJECTORY_RESULT_CANCELED:
+      report = "canceled";
+      break;
+    case control::TrajectoryResult::TRAJECTORY_RESULT_FAILURE:
+    default:
+      report = "failure";
+      break;
+  }
+  std::cout << "\033[1;32mTrajectory report: " << report << "\033[0m\n" << std::endl;
+}
+
+int main(int argc, char* argv[])
+{
+  urcl::setLogLevel(urcl::LogLevel::INFO);
+
+  // Parse the ip arguments if given
+  std::string robot_ip = DEFAULT_ROBOT_IP;
+  if (argc > 1)
+  {
+    robot_ip = std::string(argv[1]);
+  }
+
+  // Making the robot ready for the program by:
+  // Connect the the robot Dashboard
+  g_my_dashboard.reset(new DashboardClient(robot_ip));
+  if (!g_my_dashboard->connect())
+  {
+    URCL_LOG_ERROR("Could not connect to dashboard");
+    return 1;
+  }
+
+  // Stop program, if there is one running
+  if (!g_my_dashboard->commandStop())
+  {
+    URCL_LOG_ERROR("Could not send stop program command");
+    return 1;
+  }
+
+  // Power it off
+  if (!g_my_dashboard->commandPowerOff())
+  {
+    URCL_LOG_ERROR("Could not send Power off command");
+    return 1;
+  }
+
+  // Power it on
+  if (!g_my_dashboard->commandPowerOn())
+  {
+    URCL_LOG_ERROR("Could not send Power on command");
+    return 1;
+  }
```

Comment:
What exactly is the motivation behind this? When doing tests with a real robot that seems a bit weird that the robot switches off after starting that program. Also, it takes a lot of time to boot the robot again (which seems unnecessary). If we reduce this to only switching the robot an and releasing the brakes, it would have the same effect that we know the robot will be running afterwards without the additional switch off / switch on thing.

Reply:
I copied it from the full_driver_example. 
Think it was to ensure that same starting point. But agree it could be optimized